diff --git a/node_modules/expo-updates/android/build.gradle b/node_modules/expo-updates/android/build.gradle
index 145c2b3..de94f74 100644
--- a/node_modules/expo-updates/android/build.gradle
+++ b/node_modules/expo-updates/android/build.gradle
@@ -6,6 +6,9 @@ apply plugin: 'maven-publish'
 group = 'host.exp.exponent'
 version = '0.14.6'
 
+def ex_updates_native_debug_env = System.getenv("EX_UPDATES_NATIVE_DEBUG") ?: "0"
+def ex_updates_native_debug = ex_updates_native_debug_env == "1" ? "true" : "false"
+
 apply from: "../scripts/create-manifest-android.gradle"
 
 buildscript {
@@ -80,6 +86,7 @@ android {
     versionName '0.14.6'
     consumerProguardFiles("proguard-rules.pro")
     testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+    buildConfigField("boolean", "EX_UPDATES_NATIVE_DEBUG", ex_updates_native_debug)
     // uncomment below to export the database schema when making changes
     /* javaCompileOptions {
       annotationProcessorOptions {
diff --git a/node_modules/expo-updates/android/src/main/java/expo/modules/updates/UpdatesPackage.kt b/node_modules/expo-updates/android/src/main/java/expo/modules/updates/UpdatesPackage.kt
index df82184..76c099a 100644
--- a/node_modules/expo-updates/android/src/main/java/expo/modules/updates/UpdatesPackage.kt
+++ b/node_modules/expo-updates/android/src/main/java/expo/modules/updates/UpdatesPackage.kt
@@ -25,19 +25,21 @@ class UpdatesPackage : Package {
   }
 
   override fun createReactNativeHostHandlers(context: Context): List<ReactNativeHostHandler> {
+    val useNativeDebug = BuildConfig.EX_UPDATES_NATIVE_DEBUG;
+    Log.w("expo-updates-package", "EX_UPDATES_NATIVE_DEBUG = $useNativeDebug")
     val handler: ReactNativeHostHandler = object : ReactNativeHostHandler {
       private var mShouldAutoSetup: Boolean? = null
 
       override fun getJSBundleFile(useDeveloperSupport: Boolean): String? {
-        return if (shouldAutoSetup(context) && !useDeveloperSupport) UpdatesController.instance.launchAssetFile else null
+        return if (shouldAutoSetup(context) && (useNativeDebug || !useDeveloperSupport)) UpdatesController.instance.launchAssetFile else null
       }
 
       override fun getBundleAssetName(useDeveloperSupport: Boolean): String? {
-        return if (shouldAutoSetup(context) && !useDeveloperSupport) UpdatesController.instance.bundleAssetName else null
+        return if (shouldAutoSetup(context) && (useNativeDebug || !useDeveloperSupport)) UpdatesController.instance.bundleAssetName else null
       }
 
       override fun onWillCreateReactInstanceManager(useDeveloperSupport: Boolean) {
-        if (shouldAutoSetup(context) && !useDeveloperSupport) {
+        if (shouldAutoSetup(context) && (useNativeDebug || !useDeveloperSupport)) {
           UpdatesController.initialize(context)
         }
       }
@@ -45,7 +47,7 @@ class UpdatesPackage : Package {
       override fun onDidCreateReactInstanceManager(reactInstanceManager: ReactInstanceManager, useDeveloperSupport: Boolean) {
         // WHEN_VERSIONING_REMOVE_FROM_HERE
         // This code path breaks versioning and is not necessary for Expo Go.
-        if (shouldAutoSetup(context) && !useDeveloperSupport) {
+        if (shouldAutoSetup(context) && (useNativeDebug || !useDeveloperSupport)) {
           UpdatesController.instance.onDidCreateReactInstanceManager(reactInstanceManager)
         }
         // WHEN_VERSIONING_REMOVE_TO_HERE
