diff --git a/node_modules/expo/Expo.podspec b/node_modules/expo/Expo.podspec
index 0c2b8a7..87db680 100644
--- a/node_modules/expo/Expo.podspec
+++ b/node_modules/expo/Expo.podspec
@@ -18,4 +18,12 @@ Pod::Spec.new do |s|
   s.dependency 'ExpoModulesCore'
 
   s.source_files = 'ios/**/*.{h,m,swift}'
+
+
+  ex_updates_native_debug = ENV['EX_UPDATES_NATIVE_DEBUG'] == '1'
+  if ex_updates_native_debug
+    s.pod_target_xcconfig = {
+      'OTHER_CFLAGS[config=Debug]' => "$(inherited) -DEX_UPDATES_NATIVE_DEBUG=1" 
+    }
+  end  
 end
diff --git a/node_modules/expo/ios/EXAppDefinesLoader.m b/node_modules/expo/ios/EXAppDefinesLoader.m
index dd1ee3f..b17f703 100644
--- a/node_modules/expo/ios/EXAppDefinesLoader.m
+++ b/node_modules/expo/ios/EXAppDefinesLoader.m
@@ -16,6 +16,11 @@
     @"APP_DEBUG": @(YES),
 #else
     @"APP_DEBUG": @(NO),
+#endif
+#if EX_UPDATES_NATIVE_DEBUG
+    @"APP_UPDATES_DEBUG": @(YES),
+#else
+    @"APP_UPDATES_DEBUG": @(NO),
 #endif
     @"APP_RCT_DEBUG": @(RCT_DEBUG),
     @"APP_RCT_DEV": @(RCT_DEV),
diff --git a/node_modules/expo/scripts/autolinking.rb b/node_modules/expo/scripts/autolinking.rb
index 0617de5..78d67ea 100644
--- a/node_modules/expo/scripts/autolinking.rb
+++ b/node_modules/expo/scripts/autolinking.rb
@@ -30,4 +30,19 @@ def expo_patch_react_imports!(installer, options = {})
   end
 
   Expo::ReactImportPatcher.new(installer, options).run!
+  expo_patch_for_updates_debug!(installer)
+end
+
+def expo_patch_for_updates_debug!(installer)
+  projects = installer.aggregate_targets
+    .map{ |t| t.user_project }
+    .uniq{ |p| p.path }
+    .push(installer.pods_project)
+
+  projects.each do |project|
+    project.build_configurations.each do |config|
+      config.build_settings['OTHER_CFLAGS'] = "$(inherited) -DEX_UPDATES_NATIVE_DEBUG=1"
+    end
+    project.save()
+  end
 end
